### DISCUSSION TITLE
Is there a simple way to whitelist a file with .airflowignore?

### DISCUSSION THREAD

**Discussion by leopiccaro** [apache/airflow#37011] [Q&A] [answered]
I'm using .airflowignore in the re2 state (unfortunately I can't use glob), and all I'm trying to do is to ignore every single file except a specific one (let's say, dag_1.py).

All the examples I find are focused on ignoring specific files, not the other way around it. 
I added a simple regex on the .airflowignore:

```
# Ignoring everything except dag_1 files
/^(?!.*dag_1).*$/
```

But it's not working as supposed to. 
Is it a matter of the regex or something that we can't do yet?

Thanks for you time.

**Comment by Sirivasv** [comment, marked as answer]
It seems to be a matter of the syntax of the regex.
Try with:
```
# Ignoring everything except dag_1 files
^(?!.*dag_1).*$
```

 Apparently .airflowignore uses pythons `re.findall()` from the [docs](https://airflow.apache.org/docs/apache-airflow/1.10.9/concepts.html#airflowignore). And re accepts the following expressions: https://docs.python.org/3/library/re.html
 
 In their examples I failed to find one with `/`. And doing a local test in python it worked like above.

**Reply by leopiccaro** [reply]
Thanks for the clarification! 
I changed the structure and it worked. 

Just one final question, on the documentation is stated: 
(If a directory’s name matches any of the patterns, this directory and all its subfolders would not be scanned by Airflow at all. This improves efficiency of DAG finding).

When I test it here, I found out that if the file is in a subfolder inside the subfolder of where the .airflowignore is, it doesn't work. It seems that the airflowignore only goes one subfolder in, is that right?

**Reply by Sirivasv** [reply]
Indeed but I would think a modification on the regex would allow identify the ones in deeper subfolders, maybe something in the lines of:
```
^(.*/)*(?!.*dag_1).*$
```

**Reply by leopiccaro** [reply]
For some reason this specific regex is not working here, but thanks for the tip!

I don't know if the airflow ignore itself goes deeper than one level, so my current fix is to have two .airflowignore, one located at the first folder (which ignores everything expect the dag folder) and another on the dag folder (that ignores everything except the important dag files).

**Reply by Sirivasv** [reply]
I think that your initial assumption is correct once that the directory matches the regex to be ignored it stops searching deeper and thus the workaround is to have one airflowignore per folder as you are doing 👍

**Reply by leopiccaro** [reply]
Yeah, I think an interesting fix on the documentation is to leave this information more clear, but the most important thing is that I was able to correct this. Thank you so much for your help!

**Reply by potiuk** [reply]
FYI PR with documentation improvements are good first contributions and they are most welcome.
